packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -0,0 +1,59 @@
+import { writeFile } from 'fs/promises';
+import * as chokidar from 'chokidar';
+
+import { ValidationError } from '../../errors';
+import Configuration from '../../configuration/types/configuration';
+
+import { formatReport } from './formatReport';
+import { default as buildScanner } from './scanner';
+import { exists } from 'fs';
+import { promisify } from 'util';
+
+type WatchScanOptions = {
+  appmapDir: string;
+  configData: Configuration;
+};
+
+export class Watcher {
+  watcher?: chokidar.FSWatcher;
+
+  constructor(private options: WatchScanOptions) {}
+
+  async watch(): Promise<void> {
+    this.watcher = chokidar.watch(`${this.options.appmapDir}/**/mtime`, {
+      ignoreInitial: true,
+    });
+    this.watcher.on('add', this.scan.bind(this)).on('change', this.scan.bind(this));
+  }
+
+  abort(): void {
+    if (!this.watcher) return;
+
+    this.watcher.close();
+    this.watcher = undefined;
+  }
+
+  async scan(fileName: string): Promise<void> {
+    const pathTokens = fileName.split('/');
+    const appmapDir = pathTokens.slice(0, pathTokens.length - 1).join('/');
+    const appmapFile = [appmapDir, 'appmap.json'].join('.');
+    const reportFile = [appmapDir, 'appmap-findings.json'].join('/');
+
+    if (!(await promisify(exists)(appmapFile))) return;
+
+    const scanner = await buildScanner(true, this.options.configData, [appmapFile]).catch(
+      (error: Error) => {
+        throw new ValidationError(error.message + '\nUse --all to perform an offline scan.');
```

Comment:
This conflicts with an error above stating not to use `--all` with `--watch`. As it stands now, there doesn't seem to be a way to run with `--watch` completely offline.
```
$ yarn start scan --watch -d ~/dev/applandinc/appland/tmp/appmap        
Unexpected status code: 401
```

Reply:
Because `reportAllFindings` is hard-coded to `true` here, it will always be offline. I'm not sure why it didn't work for you. I've removed the catch clause since it should never be needed.